Create opcua-1021-5-0-new-received-timestamp-in-ms-placeholder-for-opc-ua-custom-action.md (#3130)
SoftwareAG/c8y-docs: content/change-logs/device-management/opcua-1021-5-0-new-received-timestamp-in-ms-placeholder-for-opc-ua-custom-action.md

---

> date: ""

Reviewer: ```suggestion
date: 2025-04-03
```

title: New receivedTimestampInMs placeholder for OPC UA custom action requests
product_area: Device management & connectivity
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-4213
version: 1021.5.0
---

> For OPC UA custom action requests, you can now include the timestamp of when the node value is received by the OPC UA device gateway. Simply add the ${receivedTimestampInMs} placeholder to the custom action body template in the OPC UA device protocol. This enables users to track when a node value is processed within the gateway.

Reviewer: ```suggestion
For OPC UA custom action requests, you can now include the timestamp of when the node value is received by the OPC UA device gateway. Simply add the `${receivedTimestampInMs}` placeholder to the custom action body template in the OPC UA device protocol. This enables users to track when a node value is processed within the gateway.
```

> For OPC UA custom action requests, the timestamp when the node value is received by the OPC UA device gateway can now be added to the request body by adding the `${receivedTimestampInMs}` placeholder to the variable's custom action body template in the OPC UA device protocol. This allows users to track when a node value is processed within the OPC UA device gateway.

Reviewer 2: The first sentence is long and complicated - here's CGPT suggestion:
"For OPC UA custom action requests, you can now include the timestamp of when the node value is received by the OPC UA device gateway. Simply add the ${receivedTimestampInMs} placeholder to the custom action body template in the OPC UA device protocol. This enables users to track when a node value is processed within the gateway."

> For OPC UA custom action requests, the timestamp when the node value is received by the OPC UA device gateway can now be added to the request body by adding the `${receivedTimestampInMs}` placeholder to the variable's custom action body template in the OPC UA device protocol. This allows users to track when a node value is processed within the OPC UA device gateway.

Reviewer: ```suggestion
For OPC UA custom action requests, you can now include the timestamp of when the node value is received by the OPC UA device gateway. Simply add the ${receivedTimestampInMs} placeholder to the custom action body template in the OPC UA device protocol. This enables users to track when a node value is processed within the gateway.
```

> Now for the OPC UA custom action requests, the timestamp when the node value is received by OPC UA device gateway can be added to the request body. This can be done by adding ${receivedTimestampInMs} placeholder to variable's custom action body template in OPC UA device protocol. This allows users to track when a node value is processed within the OPC UA device gateway.

Reviewer 3: ```suggestion
For OPC UA custom action requests, the timestamp when the node value is received by the OPC UA device gateway can now be added to the request body by adding the `${receivedTimestampInMs}` placeholder to the variable's custom action body template in the OPC UA device protocol. This allows users to track when a node value is processed within the OPC UA device gateway.
```
